### ISSUE TITLE
Stop nagging user about GPS suatus.

### ISSUE THREAD

**Issue by githubbantor** [mapsme/omim#5298]
Everytime when I'm using MapsMe, it nags me about "turn GPS on".
Would you please add an option to settings, to disable this nagging completely?

I will enable GPS turn on when I really need to do so.
This is not batery friendly, and asking too many times makes me angry.

**Comment by githubbantor** [comment]
X about GPS suatus.
O about GPS status.

**Comment by Codain** [comment]
+1
I often use Maps.ME without the GPS and whenever I switch to an already opened app instance I have this message. Plus, if I shut off the display and reenable few minutes later I may have 10 messages like this superimposed. No use to display 10 times the same message at the same time.

**Comment by Alexuser1** [comment]
@Codain Hello!

Please answer on questions:
1) What MAPS.ME version with this bug?
2) What device and os version?
3) What location settings: location off / safe mode?

Thanks!

**Comment by Codain** [comment]
@Alexuser1
I use version 7.1.3-Google on an Android 6.0.1 BlackBerry Priv device.

Location is off when I have those messages. I turn on only when I need to use the routing function. I don't know what you mean by safe mode so I guess I'm not using it.

**Comment by Alexuser1** [comment]
Hello @Codain !

Unfortunately, we don't have BB Priv and we couldn't reproduce this issue on any of our test devices on Android 6.0.1 and other versions. Please record screencast via AZ screen recorder and send us to bugs@maps.me specify the number of the issue in the email subject.

Thanks!

**Comment by he7d3r** [comment]
:+1:

**Comment by jaller94** [comment]
A message like the one described is bugging me as well.

MAPS.ME version 7.1.3-Google (com.mapswithme.maps.pro)
Permissions given: Contacts (always), Locations (always), Storage (always)
Location mode: Location off (no GPS & no Google services)
OS: Cyanogenmod 13 (Android 6.0.1)
Device: Google/LG Nexus 4 (mako)

[Screenshot on Google Drive as long as Github upload is broken for me](https://drive.google.com/open?id=0B9HSrppmOV_seGpKQmxScFJ1b3M)

User workaround: Deny Permission "Location" to MAPS.ME

**Comment by mortalis13** [comment]
Having the same problem.
With the GPS disabled shows a message asking to turn it on every time the app starts.
Need a setting to disable location service in the app.
Thanks.

**Comment by thany** [comment]
This setting used to be available. The location button (the round button in the lower left) used to be a tristate button (follow location, follow location & compass, don't follow). This functionality has been deliberately removed, and I've been asking to put it back ever since, because no working alternative has been put into place, which is frankly just negligent of the devs.

**Comment by davebrny** [comment]
this alert is starting to get on my nerves me too. it seems unnecessary anyway because if the gps is off, you will figure out in a second or two when their position doesnt show up on the map. 

maybe the location button under the zoom controls would be a better place to let the user know? 
this is already the button you press when you want to center the map on your location and if the gps is trying to lock it also shows a radar animation. so why not show a red icon here if gps is not turned on, and if you click it _then_ it shows this alert dialog with the option to go to location settings or else just go there directly when clicked

**Comment by davebrny** [comment]
i made a tasker profile that closes this message if anyone is interested (you need to have the autoinput plugin installed for it to work though)  
https://gist.github.com/davebrny/349259436121d0833f64bbdff2dac965

**Comment by thany** [comment]
Please try not to solve this with workarounds. THis will only further deter the maps.me devs from fixing it.

Even though this problem has existed for literally YEARS, but one can always be hopeful that someone might get around to spending 5 minutes to fix this. Who knows.

**Comment by davebrny** [comment]
ok @thany ill try better next time not to solve any problems that are annoying me and i apologise to anyone here who wanted this to be fixed but i guess thats it now... the map.me devs have taken the rest of the year off since that they dont have to figure out how to stop this message from popping up

**Comment by thany** [comment]
It was mostly a rante towards the MAPS.ME devs, @davebrny. Sorry if I offended you.

It's just that issue has existed for literally years. Seriously. It has been reported, and reported, and reported. Over and over and over again. They just won't fix it, it it literally is a 5 minute fix. If only I had the expertise.

**Comment by trutypesounds** [comment]
Yes please, this feature is hell annoying, let me try to explain - when you are on a bicycle trip with tent (so no hotels, wifi/lte anything like that) in a country new to you, you try to preserve battery as much as u can, so my phone usually full save mode, gps only mode, and every time i unlock the screen to see where i am - BAM! - the app pops up a dialog asking the user to enable location access, imaging pushing the cancel/stop button when u r on bicycle down a mountain serpantine going full speed with cars and motorcycles passing by! Please, make this an option (on/off)!!

**Comment by thany** [comment]
> imaging pushing the cancel/stop button when u r on bicycle down a mountain serpantine going full speed with cars and motorcycles passing by! Please, make this an option (on/off)!!

In my country that's even illegal without stopping first. Cyclists are not allowed to use a mobile phone while moving.

Regardless, the app just simply doesn't respect my setting for not allowing it location access. I don't see why fixing this has to take so long.

MAPS.ME devs: what do you need to get this fixed in the next, say, 4 weeks or so? Should be doable.

**Comment by trutypesounds** [comment]
> > imaging pushing the cancel/stop button when u r on bicycle down a mountain serpantine going full speed with cars and motorcycles passing by! Please, make this an option (on/off)!!
> 
> In my country that's even illegal without stopping first. Cyclists are not allowed to use a mobile phone while moving.
> 
> Regardless, the app just simply doesn't respect my setting for not allowing it location access. I don't see why fixing this has to take so long.
> 
> MAPS.ME devs: what do you need to get this fixed in the next, say, 4 weeks or so? Should be doable.

had no issues using phone as a gps while moving on a bicycle in northern europe, anyway using the phone in a cradle mounted on a wheelbar. when my route is set i just want to look at the map from time to time that im on the right way, when i need to search something specific or correct the route thats for sure i need to stop at the safe place of the road and do this, never when on a move.

**Comment by Codain** [comment]
I found a solution some years ago: I replaced MapsMe app by a concurrent one and am no longer annoyed. Not sure it is a valid solution here but at least I'm happy. This issue was not the only driver (there was other points) but had a huge weight on the balance as soon as I use more often use map with no navigation than with navigation.
